launcher/src/main.rs

```diff
@@ -18,30 +18,131 @@ use std::sync::{mpsc, Arc};
 use std::thread;
 use std::thread::sleep;
 use std::time::{Duration, Instant};
-use std::{fs, io};
+use std::{
+    fs, io,
+    io::{Read, Write},
+};
 use thiserror::Error;
 use tracing_subscriber::{filter::LevelFilter, EnvFilter};
 
 mod env_runtime;
 
+fn get_config(
```

Comment:
Same, this could be done through `Info` in the future.

Reply:
I think I prefer doing it super early on, so that dependencies between flags can be determined early, and the regular code can just use 1 strong value.

But I see the point.